app/lib/service/services.dart

```diff
@@ -44,16 +45,23 @@ import 'spam/backend.dart';
 ///  * storage wrapped with retry.
 Future<void> withServices(FutureOr<void> Function() fn) async {
   return withAppEngineServices(() async {
-    return await withStorageRetry(() async {
-      return await withPubServices(fn);
-    });
+    return await withPubServices(fn);
   });
 }
 
 /// Run [fn] with pub services that are shared between server instances, CLI
 /// tools and integration tests.
 Future<void> withPubServices(FutureOr<void> Function() fn) async {
   return fork(() async {
+    if (activeConfiguration.usesRealAppEngine) {
+      final authClient = await auth
+          .clientViaApplicationDefaultCredentials(scopes: [...Storage.SCOPES]);
+      final storageClient = httpRetryClient(innerClient: authClient);
+      registerStorageService(
+          Storage(storageClient, activeConfiguration.projectId));
+      registerScopeExitCallback(() async => storageClient.close());
+    }
```

Comment:
Doesn't this need an `else` branch? to register some other `storageClient`?

Should we consider calling the configuration option something like `Configuration.useFakeServices` or? Don't we need a similar way to configure that we want to use the fake datastore, the fake email sender, the fake cloud storage, the fake signature signer?

Reply:
> Doesn't this need an else branch? to register some other storageClient

In the cases where `pkg/fake_gcloud` we register a custom `storageService` and `dbService`. In such case there is no reason to call  `auth. clientViaApplicationDefaultCredentials`, and we don't have HTTP client either to wrap in retrying HTTP client.